google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleFrameworkDetector.java

```diff
@@ -66,8 +76,37 @@ public FileType getFileType() {
   @NotNull
   @Override
   public ElementPattern<FileContent> createSuitableFilePattern() {
-    return StandardPatterns.or(
-        FileContentPattern.fileContent().withName("app.yaml"),
-        FileContentPattern.fileContent().withName("app.yml"));
+    return new AppEngineFlexPattern().withAppEngineFlexYAMLContent();
+  }
+
+  /**
+   * IntelliJ API pattern class that checks for App Engine Flex project file presence and checks it
+   * has required configuration lines to avoid spurious detection.
+   */
+  private static class AppEngineFlexPattern
+      extends ObjectPattern<FileContent, AppEngineFlexPattern> {
+
+    private AppEngineFlexPattern() {
+      super(FileContent.class);
+    }
+
+    private AppEngineFlexPattern withAppEngineFlexYAMLContent() {
+      return with(
+          new PatternCondition<FileContent>("with-appengine-java-flexible") {
+            @Override
+            public boolean accepts(@NotNull FileContent fileContent, ProcessingContext context) {
+              // checks for flex engine file names and then checks for required configuration line
+              // inside.
+              boolean nameMatch = APP_ENGINE_FLEX_PROJECT_FILES.contains(fileContent.getFileName());
+              if (nameMatch) {
+                Scanner scanner = new Scanner(fileContent.getContentAsText().toString());
+                while (scanner.hasNextLine()) {
+                  if (scanner.nextLine().startsWith(APP_ENGINE_REQUIRED_YAML)) return true;
```

Comment:
@nkibler I'm personally ok with it, but does google style guid allow {}'less statements like this?

Reply:
It does not, sorry! http://go/java-style#s4.1.1-braces-always-used